File: backend.Domain/Entities/ArabicLetter.cs
namespace backend.Domain.Entities;

public class ArabicLetter
{
    public Guid Id { get; set; }
    public int SequenceOrder { get; set; }
    public string Character { get; set; } = string.Empty;
    public string NameArabic { get; set; } = string.Empty;
    public LocalizedText Name { get; set; } = null!;
    public LocalizedText Transliteration { get; set; } = null!;
    public string MakhrajType { get; set; } = string.Empty;
    public LocalizedText MakhrajDescription { get; set; } = null!;
    public string Sifaat { get; set; } = string.Empty;
    public string ExampleWordArabic { get; set; } = string.Empty;
    public LocalizedText ExampleWordMeaning { get; set; } = null!;
    public string IsolatedForm { get; set; } = string.Empty;
    public string InitialForm { get; set; } = string.Empty;
    public string MedialForm { get; set; } = string.Empty;
    public string FinalForm { get; set; } = string.Empty;
    public bool IsConnector { get; set; } = true;

    public ICollection<LetterHarakatItem> HarakatItems { get; set; } = [];
}

File: android/app/src/main/java/com/example/quran_app/domain/model/ArabicLetter.kt
package com.example.quran_app.domain.model

import com.google.gson.annotations.SerializedName

data class ArabicLetter(
    @SerializedName("id")                  val id: String,
    @SerializedName("order")               val order: Int,
    @SerializedName("letter")              val letter: String,
    @SerializedName("nameEnglish")         val nameEnglish: String,
    @SerializedName("nameArabic")          val nameArabic: String,
    @SerializedName("nameBangla")          val nameBangla: String,
    @SerializedName("transliteration")     val transliteration: String,
    @SerializedName("makhrajType")         val makhrajType: String,
    @SerializedName("makhrajDescription")  val makhrajDescription: String,
    @SerializedName("makhrajDescriptionBn")val makhrajDescriptionBn: String,
    @SerializedName("sifaat")              val sifaat: List<String>,
    @SerializedName("exampleWordArabic")   val exampleWordArabic: String,
    @SerializedName("exampleWord")         val exampleWord: String,
    @SerializedName("exampleWordBn")       val exampleWordBn: String,
    @SerializedName("isolatedForm")        val isolatedForm: String,
    @SerializedName("initialForm")         val initialForm: String,
    @SerializedName("medialForm")          val medialForm: String,
    @SerializedName("finalForm")           val finalForm: String,
    @SerializedName("isConnector")         val isConnector: Boolean,
    @SerializedName("audioUrl")            val audioUrl: String
)
